hack/testing/rsyslog/roles/rsyslog/tasks/main.yaml

```diff
@@ -0,0 +1,175 @@
+---
+- name: Install required packages
+  yum:
+    name: '{{ item }}'
+    state: 'present'
+  with_flattened:
+    - '{{ rsyslog__viaq_prereq_packages }}'
+    - '{{ rsyslog__base_packages }}'
+    - '{{ rsyslog__viaq_packages if rsyslog__viaq|bool else [] }}'
+    - '{{ rsyslog__tls_packages if rsyslog__pki|bool else [] }}'
+    - '{{ rsyslog__packages }}'
+  when: rsyslog__enabled|bool
+
+- name: Create required system group
+  group:
+    name: '{{ rsyslog__group }}'
+    state: 'present'
+    system: True
+  when: rsyslog__unprivileged|bool and rsyslog__group != 'root'
+
+- name: Create required system user
+  user:
+    name: '{{ rsyslog__user }}'
+    group: '{{ rsyslog__group }}'
+    home: '{{ rsyslog__home }}'
+    shell: '/bin/false'
+    state: 'present'
+    createhome: False
+    system: True
+  when: rsyslog__unprivileged|bool and rsyslog__user != 'root'
+
+# Creating a work directory
+- name: Create a work directory
+  file:
+    state: directory
+    path: "{{ rsyslog__work_directory }}"
+    mode: 0700
```

Comment:
should add `owner` and `group` here, then I'm not sure if you need the spool dir task below

Reply:
I've added ```owner``` and ```group```.

It seems /var/spool/rsyslog used to be used for a WorkDirectory.
```$WorkDirectory /var/spool/rsyslog  # default location for work (spool) files```
We use ```/var/lib/rsyslog``` by default, which is owned by the rsyslog package:
```
$ rpm -qf /var/lib/rsyslog
rsyslog-8.24.0-33.el7.x86_64
```
Note: we created the work directory (rsyslog__work_dir) only when the owner of rsyslog was not root. 
``` 
when: rsyslog__unprivileged|bool and rsyslog__user != 'root'
```
I think we should not have the condition here since if I deploy rsyslog as a root and set the work dir to somewhere else like /var/spool/rsyslog, which won't be created...

Reply:
> It seems /var/spool/rsyslog used to be used for a WorkDirectory.

That's a debian-ism - default on rhel/fedora/centos is `/var/lib/rsyslog`

> We use /var/lib/rsyslog by default, which is owned by the rsyslog package:

So by default, we should use the same mode, owner, and group that the rsyslog package uses.  That way you will get exactly the same thing if you run the ansible playbook after installing the rpm rsyslog package.

Reply:
Yep, we don't change the mode, owner and group of the directory.